src/server/init-component.js

```diff
@@ -0,0 +1,17 @@
+export function initComponent (component) {
```

Comment:
While this fixes the hook invocation order, this is mounting the entire component tree synchronously. I'm wondering if it's possible to delay the mounting to play nicer with streaming mode...